helper/schema/resource.go

```diff
@@ -1482,3 +1482,135 @@ func RemoveFromState(d *ResourceData, _ interface{}) error {
 	d.SetId("")
 	return nil
 }
+
+// Internal validation of provider implementation
+func (r *ResourceIdentity) InternalIdentityValidate() error {
+	if r == nil {
+		return fmt.Errorf(`The resource identity is empty`)
+	}
+
+	if r.Schema == nil {
+		return fmt.Errorf(`The resource identity schema is empty`)
+	}
```

Comment:
I think we should also check whether the length of the Schema is zero and return an error that at least one attribute is required. What do you think?